### DISCUSSION TITLE
Accessing audio format, bit depth, and sample rate as track fields?

### DISCUSSION THREAD

**Discussion by ali-ramadhan** [MoeMusic/Moe#221] [Ideas]
### Describe the new feature

I would like to use the file format, bit depth, and sample rate of tracks to format paths. I see that `audio_format` is a `Track` property (but not a field) and that bit depth and sample rate are not properties nor fields.

I was wondering if it would make sense to add bit depth and sample rate as `Track` properties, then make all three fields so they could be used to format paths?

They are accessible via the `mediafile` I think , so I could try to add them in as `Track` properties via `@property`.

I think it's common to use audio format, bit depth, and sample rate to format album paths, however it might not make sense for them to be album properties. One reason being that some albums (some digital releases?) could have a mixture of formats, bit depths, and sampling rates.

So to format album paths using this information, it might make sense to create little plugins to return e.g. `album_audio_format` which could be `FLAC` or `MP3` or even `FLAC+MP3`.

But this would still require audio format, bit depth, and sample rate to be `Track` properties I think.

**Comment by jtpavlock** [comment]
Made this a discussion since there's a lot to unpack here and I'd like to keep issues to single, actionable items.

So despite the docs, path formats actually use the class attributes. So you can currently use `track.audio_format` in your path if you'd like. The reason it wasn't in the fields section of the docs is because you can't query or edit it. However, I should probably include it anyways with footnotes that it just can't be queried or edited to avoid confusion with the path formats. Ideally, an end-user doesn't have to look at the api docs at all unless they're creating a plugin or something.

Per the album properties, I agree it doesn't really make a lot of sense for those fields to be album properties, and may be better off as custom plugin functions. If I see there's a common demand for those fields as path template functions it may be worth adding them into core, but I'd like to hold off on that for now I think.

So as far as actionable items from this, I think we have:

1. Add all track and album attributes to field docs
2. Add bit depth property/field to a track
3. Add sample rate property/field to a track

I just merged #222 for the docs fix, but feel free to create feature requests for or implement 2 or 3.
